src/analysis/souffle/tags.dl

```diff
@@ -28,4 +28,21 @@
 .type Tag <: symbol
 .decl isTag(tag: Tag)
 
+// An access path marked with an integrity tag (shortened here to IntegTag)
+// indicates that some property must be true of the data at the given
+// AccessPath. The conservative over-approximation for propagating an integrity
+// tag is that a node has an integrity tag only if all of its predecessors have
+// that integrity tag. Checking this property requires an intersection, which
+// unfortunately cannot be directly expressed in datalog.
+.type IntegTag <: symbol
```

Comment:
Let us call it `IntegrityTag`?

Reply:
Fixed.

Reply:
Leaving distinction for now, as mentioned elsewhere, and will remove if we get unwanted friction.